### PR TITLE
fix(spans): SQL comments & whitespace

### PR DESCRIPTION
**Pull request by jjbayer**
For example,

```sql
    select * -- some comment
            from (
                x
            )
 ```

becomes

```sql
select * from (x)
```

#skip-changelog

### CODE REVIEW

File: relay-general/src/store/normalize/span/description.rs [hunk 1]
```diff
@@ -534,28 +537,28 @@ mod tests {
 
     span_description_test!(
         span_description_dont_scrub_double_quoted_strings_format_postgres,
-        r#"SELECT * from \"table\" WHERE sku = %s"#,
+        r#"SELECT * from "table" WHERE sku = %s"#,
```
Comment (jjbayer) [inline relay-general/src/store/normalize/span/description.rs:548]: `span_description_test!` previously expected JSON encoded strings, hence the escape characters. Now it takes pure strings.

File: relay-general/src/store/normalize/span/description.rs [hunk 2]
```diff
@@ -13,6 +13,8 @@ use crate::store::span::tag_extraction::HTTP_METHOD_EXTRACTOR_REGEX;
 use crate::store::SpanDescriptionRule;
 use crate::types::{Annotated, ProcessingResult, Remark, RemarkType, Value};
 
+static SQL_COMMENTS: Lazy<Regex> = Lazy::new(|| Regex::new(r"--.*(?P<newline>\n)").unwrap());
```
Comment (iker-barriocanal) [inline relay-general/src/store/normalize/span/description.rs:17]: nit: should we add docstrings?

File: relay-general/src/store/normalize/span/description.rs [hunk 3]
```diff
@@ -36,6 +38,13 @@ static SQL_NORMALIZER_REGEX: Lazy<Regex> = Lazy::new(|| {
     .unwrap()
 });
 
+/// Removes extra whitespace and newlines.
+static SQL_WHITESPACE: Lazy<Regex> = Lazy::new(|| Regex::new(r"(\s*\n\s*)|(\s\s+)").unwrap());
+
+/// Removes whitespace around parentheses.
+static SQL_PARENS: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"((?P<pre>\()\s+)|(\s+(?P<post>\)))").unwrap());
```
Comment (iker-barriocanal) [inline relay-general/src/store/normalize/span/description.rs:47]: Q: do we need the capture group names?

**Review by iker-barriocanal — APPROVED** [review]

File: relay-general/src/store/normalize/span/description.rs (hunk 2 above)
Comment (jjbayer) [inline relay-general/src/store/normalize/span/description.rs:17]:
```suggestion
/// Removes SQL comments starting with "--".
static SQL_COMMENTS: Lazy<Regex> = Lazy::new(|| Regex::new(r"--.*(?P<newline>\n)").unwrap());
```

File: relay-general/src/store/normalize/span/description.rs (hunk 3 above)
Comment (jjbayer) [reply relay-general/src/store/normalize/span/description.rs:47]: `scrub_sql_queries` uses them for replacement, so I would say yes.

**Pull request merged**